### PR TITLE
Updated StackDriver and Flexible Env documentation

### PR DESCRIPTION
**Pull request by matthewloring**

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots PTAL.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Google Cloud Trace for Node.js
+# StackDriver Trace for Node.js
```
Comment (ofrobots) [inline README.md]: Should it be Google StackDriver Trace?

Comment (matthewloring) [inline README.md]: @justinbeckwith Could we get some branding guidance here.

File: README.md [hunk 2]
```diff
@@ -42,9 +42,9 @@ Alternatively, you can provide configuration through a config file. This can be
 
 There are three different services that can host Node.js application to Google Cloud Platform.
 
-### Google App Engine Managed VMs
+### Google App Engine Flexible Environment
```
Comment (JustinBeckwith) [inline README.md]: stupid nit, but we use lower case on `flexible environment`

File: README.md (hunk 1 above)
Comment (JustinBeckwith) [inline README.md]: 'StackDriver Trace` appears to be correct looking at their marketing materials.  :shrugs:

**Comment by JustinBeckwith** [comment]
LGTM

**Pull request merged**